android/java/org/chromium/chrome/browser/crypto_wallet/fragments/EditVisibleAssetsBottomSheetDialogFragment.java

```diff
@@ -175,36 +180,100 @@ public void setupDialog(@NonNull Dialog dialog, int style) {
+                @Override
+                public void onClick(View clickView) {
+                    Dialog dialog = new Dialog(getActivity());
+                    dialog.setContentView(R.layout.brave_wallet_add_custom_asset);
+                    dialog.show();
+
+                    Button cancel = dialog.findViewById(R.id.cancel);
+                    cancel.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            dialog.dismiss();
+                        }
+                    });
+                    Button add = dialog.findViewById(R.id.add);
+                    filterAddCustomAssetTextWatcher.setDialog(dialog);
+                    EditText tokenNameEdit = dialog.findViewById(R.id.token_name);
+                    EditText tokenContractAddressEdit =
+                            dialog.findViewById(R.id.token_contract_address);
+                    EditText tokenSymbolEdit = dialog.findViewById(R.id.token_symbol);
+                    EditText tokenDecimalsEdit = dialog.findViewById(R.id.token_decimals);
+                    tokenNameEdit.addTextChangedListener(filterAddCustomAssetTextWatcher);
+                    tokenContractAddressEdit.addTextChangedListener(
+                            filterAddCustomAssetTextWatcher);
+                    tokenSymbolEdit.addTextChangedListener(filterAddCustomAssetTextWatcher);
+                    tokenDecimalsEdit.addTextChangedListener(filterAddCustomAssetTextWatcher);
+                    add.setOnClickListener(new View.OnClickListener() {
+                        @Override
+                        public void onClick(View v) {
+                            BraveWalletService braveWalletService = getBraveWalletService();
+                            assert braveWalletService != null;
+                            ErcToken token = new ErcToken();
+                            token.contractAddress = tokenContractAddressEdit.getText().toString();
+                            token.name = tokenNameEdit.getText().toString();
+                            token.logo = "";
+                            token.isErc20 = true;
+                            // We support Erc20 tokens only for now
+                            token.isErc721 = false;
+                            token.symbol = tokenSymbolEdit.getText().toString();
+                            token.decimals = 18;
+                            try {
+                                token.decimals =
+                                        Integer.valueOf(tokenDecimalsEdit.getText().toString());
+                            } catch (NumberFormatException exc) {
+                            }
+                            token.visible = true;
+                            braveWalletService.addUserAsset(token, mChainId, success -> {
+                                if (success) {
```

Comment:
how it behaves when `token.name`, `token.symbol` are empty? is it handled by `braveWalletService.addUserAsset`?

Reply:
sorry, I see it is done in https://github.com/brave/brave-core/pull/11463/files#diff-9c0d546aefd39f49ed238ebfa7d1fa6f133dcfe8bd63e7841d617bee1049ac42R295